DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/AdminFieldsController.cs

```diff
@@ -211,10 +225,12 @@ public IActionResult AddAdminFieldAnswersBulk(int customisationId)
         }
 
         [HttpPost]
-        [Route("{customisationId}/AdminFields/Add/Bulk")]
+        [Route("{customisationId:int}/AdminFields/Add/Bulk")]
+        [ServiceFilter(typeof(VerifyAdminUserCanAccessCourse))]
         [ServiceFilter(typeof(RedirectEmptySessionData<AddAdminFieldData>))]
         public IActionResult AddAdminFieldAnswersBulk(
-            BulkAdminFieldAnswersViewModel model
+            BulkAdminFieldAnswersViewModel model,
+            int customisationId
         )

```

Comment:
There are a lot of cases where model holds the customisation ID and it's in the url. In that case, I (as a malicious user) could change the url to have customisation ID that I'm allowed to change, but submit a model that has a customisation ID that I'm not allowed to change. I'd be able to access the page and then be able to manage a customisation with an ID that's in the model, the one that I don't have permissions to manage. I think the best thing going forward is only keeping the customisation ID in one place - preferrably in the URL and we'd remove it from the model. If that's not possible for some reason we should at least check that these IDs are the same.

Reply:
Removed customisation Id from all the admin fields models and removed promptNumber from the editing journey ones